front/src/modules/timesStops/TimesStops.tsx

```diff
@@ -1,63 +1,43 @@
 import React, { useState, useEffect } from 'react';
 
-import { DynamicDataSheetGrid } from 'react-datasheet-grid';
+import cx from 'classnames';
+import { DynamicDataSheetGrid, type DataSheetGridProps } from 'react-datasheet-grid';
 import { useTranslation } from 'react-i18next';
 
-import type { ManageTrainSchedulePathProperties } from 'applications/operationalStudies/types';
 import { useOsrdConfActions } from 'common/osrdContext';
 import { isVia } from 'modules/pathfinding/utils';
 import type { SuggestedOP } from 'modules/trainschedule/components/ManageTrainSchedule/types';
 import type { PathStep } from 'reducers/osrdconf/types';
 import { useAppDispatch } from 'store';
-import { removeElementAtIndex } from 'utils/array';
-import { marginRegExValidation } from 'utils/physics';
 
-import { useInputColumns } from './TimeStopsColumns';
-import type { PathWaypointColumn } from './types';
+import { marginRegExValidation } from './consts';
+import { useTimeStopsColumns } from './TimeStopsColumns';
+import { TableType } from './types';
+import type { PathWaypointRow } from './types';
 import { formatSuggestedViasToRowVias } from './utils';
 
-const WITH_KP = true;
+export const WITH_KP = true;
 
 type TimesStopsProps = {
-  pathProperties: ManageTrainSchedulePathProperties | undefined;
+  allWaypoints?: SuggestedOP[];
   pathSteps?: PathStep[];
   startTime?: string;
+  tableType: TableType;
+  cellClassName?: DataSheetGridProps['cellClassName'];
+  stickyRightColumn?: DataSheetGridProps['stickyRightColumn'];
```

Comment:
You can even move the onChange function in a props:
```suggestion
  stickyRightColumn?: DataSheetGridProps['stickyRightColumn'];
  onChange?: (
    row: PathWaypointRow[],
    op: Operation,
    setRows: (rows: PathWaypointRow[]) => void
  ) => void;
```

Reply:
leaving it as discussed